event/consumer.go

```diff
@@ -36,19 +36,19 @@ func NewConsumer() *Consumer {
 }
 
 // Consume converts messages to event instances, and pass the event to the provided handler.
-func (consumer *Consumer) Consume(messageConsumer MessageConsumer, handler Handler, errorReporter reporter.ErrorReporter) {
+func (consumer *Consumer) Consume(ctx context.Context, messageConsumer MessageConsumer, handler Handler, errorReporter reporter.ErrorReporter) {
 
 	go func() {
 		defer close(consumer.closed)
 
 		for {
 			select {
-			case message := <-messageConsumer.Incoming():
+			case message := <-messageConsumer.Channels().Upstream:
 
-				processMessage(message, handler, errorReporter)
+				processMessage(ctx, message, handler, errorReporter)
```

Comment:
Looking ahead to the future I believe context will be passed through kafka messages so we can track who triggered what. Knowing this piece of information I would create a new context every time a message is received, so it is closer to the end functionality/goal.

Also it allows us to track which message resulted in a failure in kibana.

I would propose something like the following:

```
...
		for {
			select {
			case message := <-messageConsumer.Incoming():
			case message := <-messageConsumer.Channels().Upstream:

				messageCtx := context.Background()
				processMessage(message, handler, errorReporter)
				processMessage(messageCtx, message, handler, errorReporter)
...
```

Reply:
The context for the other cases in switch statement I believe are correct.